csr/csr.go

```diff
@@ -307,16 +308,35 @@ func WatchCastAINodeCSRs(ctx context.Context, log logrus.FieldLogger, client kub
 			if csrResult == nil {
 				continue
 			}
+			if csrResult.RequestingUser != "kubelet-bootstrap" {
+				log.WithFields(logrus.Fields{
+					"csr":       name,
+					"node_name": csrResult.RequestingUser,
+				}).Debugf("skipping csr not from kubelet-bootstrap: %v", csrResult.RequestingUser)
+				continue
+			}
 
 			cn, err := getSubjectCommonName(name, request)
 			if err != nil {
 				log.WithFields(logrus.Fields{
 					"csr":       name,
 					"node_name": cn,
-				}).Debugf("WatchAndApproveNodeCSRV1: skipping csr: %v", err)
+				}).Debugf("skipping csr: %v", err)
 				continue
 			}
-			if csrResult.Approved() || !isAutoApproveAllowedForNode(ctx, client, cn) {
+			if csrResult.Approved() {
+				continue
+			}
+			log.WithFields(logrus.Fields{
+				"csr":       name,
+				"node_name": cn,
+			}).Debugf("checking csr: %s, node: %s", name, cn)
+			enabled, err := isAutoApproveAllowedForNode(ctx, client, cn)
+			if !enabled || err != nil {
```

Comment:
This error handling is not idiomatic. If err is returned then the rest of parameters are undefined. Here err is checked only if enabled is false.